packages/opentelemetry-resources/src/platform/browser/detectors/BrowserDetector.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag } from '@opentelemetry/api';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { Detector, Resource, ResourceDetectionConfig } from '../../..';
+import { ResourceAttributes } from '../../../types';
+
+/**
+ * BrowserDetector will be used to detect the resources related to browser.
+ */
+class BrowserDetector implements Detector {
+  async detect(config?: ResourceDetectionConfig): Promise<Resource> {
+    const browserResource: ResourceAttributes = {
+      [SemanticResourceAttributes.PROCESS_RUNTIME_NAME]: 'browser',
```

Comment:
I think in this case the actual javascript runtime might be useful like v8 or chakra. Is this specified?

Reply:
I didn't recall any formalized methods to detect the javascript engine. Mostly the detection is based on the difference between engines of undefined behavior.

Reply:
I checked with `window,navigator` but there is no info about the underlying engine so i don't think its possible :/

Reply:
You can also detect chromium based browsers by the presence of `window.chrome` but the runtime is still an assumption.